File: predictions.py
# core/predictions.py

from .calculations import seconds_to_hms

def riegel_predict(t_seconds: float, d1: float, d2: float, exponent: float = 1.06) -> float:
    # Riegel formula: T2 = T1 * (D2 / D1)^1.06
    if d1 <= 0:
        return 0.0
    ratio = d2 / d1
    return t_seconds * (ratio ** exponent)


def format_time_hms(total_seconds: float) -> str:
    h, m, s = seconds_to_hms(total_seconds)
    return f"{h}h {m}m {s}s"


def cooper_vo2max(distance_m: float, time_seconds: float) -> float:
    # Simple Cooper-style estimate: VO2max ≈ (distance_m - 504.9) / 44.73
    # This assumes 12-minute test; here we adjust very roughly by scaling.
    if time_seconds <= 0:
        return 0.0
    # Normalize to 12 minutes
    normalized_distance = distance_m * (12 * 60 / time_seconds)
    return (normalized_distance - 504.9) / 44.73


def simple_vdot_from_pace(pace_seconds_per_km: float) -> float:
    # Very rough VDOT-like metric: faster pace => higher value
    if pace_seconds_per_km <= 0:
        return 0.0
    # Example: 4:00/km ~ VDOT 60, 5:00/km ~ VDOT 50 etc. This is a crude approximation.
    minutes = pace_seconds_per_km / 60.0
    return max(20.0, 90.0 - (minutes * 10.0))

File: predictions.java
package core;

public class Predictions {

    // --- Riegel Prediction ---
    // Riegel formula: T2 = T1 * (D2 / D1)^exponent
    public static double riegelPredict(double tSeconds, double d1, double d2, double exponent) {
        if (d1 <= 0) {
            return 0.0;
        }
        double ratio = d2 / d1;
        return tSeconds * Math.pow(ratio, exponent);
    }

    // Overload with default exponent = 1.06
    public static double riegelPredict(double tSeconds, double d1, double d2) {
        return riegelPredict(tSeconds, d1, d2, 1.06);
    }

    // --- Format time as H:M:S ---
    public static String formatTimeHms(double totalSeconds) {
        int[] hms = Calculations.secondsToHms(totalSeconds);
        int h = hms[0];
        int m = hms[1];
        int s = hms[2];
        return h + "h " + m + "m " + s + "s";
    }

    // --- Cooper VO2max Estimate ---
    // VO2max ≈ (distance_m - 504.9) / 44.73 (normalized to 12 minutes)
    public static double cooperVo2max(double distanceM, double timeSeconds) {
        if (timeSeconds <= 0) {
            return 0.0;
        }
        double normalizedDistance = distanceM * ((12 * 60.0) / timeSeconds);
        return (normalizedDistance - 504.9) / 44.73;
    }

    // --- Simple VDOT-like metric from pace ---
    public static double simpleVdotFromPace(double paceSecondsPerKm) {
        if (paceSecondsPerKm <= 0) {
            return 0.0;
        }
        double minutes = paceSecondsPerKm / 60.0;
        double estimate = 90.0 - (minutes * 10.0);
        return Math.max(20.0, estimate);
    }
}